lib/puppet/pops/types/types.rb

```diff
@@ -364,6 +402,16 @@ def eql?(o)
 # @api public
 #
 class PType < PTypeWithContainedType
+
+  def self.register_ptype(loader, ir)
+    create_ptype(loader, ir, 'AnyType',
+       'type' => {
+         KEY_TYPE => POptionalType.new(PType::DEFAULT),
+         KEY_VALUE => nil
+       }
+    )
+  end
```

Comment:
There's no Pcore::TypeWithContainedType so I didn't want to add a registration method to that class. An alternative approach could be to do that and then add an exception for it in the loop that registers all meta-types. I chose the current approach because it felt cleaner.